src/System.Xml.XDocument/src/System/Xml/Linq/Extensions.cs

```diff
@@ -374,9 +374,11 @@ static IEnumerable<XAttribute> GetAttributes(IEnumerable<XElement> source, XName
                     XContainer c = root;
                     while (true)
                     {
-                        if (c != null && c.content is XNode)
+                        if (c != null)
                         {
-                            n = ((XNode)c.content).next;
+                            XNode content = c.content as XNode;
+                            if (content != null)
+                                n = content.next;
                         }
                         else
```

Comment:
Previously, this else was taken when `c == null || !(c is XNode)`, now it is taken only when `c == null`. Why is that OK?